stl/inc/xstring

```diff
@@ -2535,13 +2567,28 @@ public:
         _Elem* _Ptr             = _My_data._Bx._Buf;
         auto&& _Alproxy         = _GET_PROXY_ALLOCATOR(_Alty, _Getal());
         _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data); // throws
-        if (_New_capacity < _New_size) {
-            _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size());
-            const pointer _Fancyptr = _Getal().allocate(_New_capacity + 1); // throws
-            _Ptr                    = _Unfancy(_Fancyptr);
+
+        bool _Activate_large_mode = _New_capacity < _New_size;
+#ifdef __cpp_lib_constexpr_string
+        _Activate_large_mode =
+            _Activate_large_mode || _STD is_constant_evaluated(); // SSO disabled in constexpr context
+#endif // __cpp_lib_constexpr_string

```

Comment:
Nitpick, I think it would be clearer to use if else here
```suggestion
#ifdef __cpp_lib_constexpr_string // SSO disabled in constexpr context
        const bool _Activate_large_mode = _New_capacity < _New_size || _STD is_constant_evaluated(); 
#else 
        const bool _Activate_large_mode = _New_capacity < _New_size;
#endif // __cpp_lib_constexpr_string
```